app/javascript/app/document-capture/components/file-input.jsx

```diff
@@ -57,7 +57,12 @@ function FileInput({ label, hint, accept, value, onChange }) {
           )}
           {!value && (
             <div className="usa-file-input__instructions" aria-hidden="true">
-              <span className="usa-file-input__drag-text">{t('doc_auth.forms.choose_file')}</span>
+              <span className="usa-file-input__drag-text">
+                {formatHTML(t('doc_auth.forms.choose_file_html'), {
+                  // eslint-disable-next-line react/prop-types
```

Comment:
This gives me pause. Technically the `u` here is a component, and the rule should apply to all components. But I'm also motivated to not over-complicate how one interfaces with `formatHTML` to impose that they define a separate component function with its own `propTypes`, `defaultProps`, etc. Conversely, I could have chosen to not assign the "handlers" mapping with the value being of type `Component`, and instead more of a generic "function which returns an element". The nice thing about this usage is that you _could_ choose to extract or reuse a traditional component here (e.g. `{ button: MyFancyButton }`).

I see some value in having the rule in general, in how it's self-documenting and provides some availability of type-hinting for capable IDEs.

But it's also something where, if we decide to integrate more of the TypeScript tooling, it's completely redundant with standard JSDoc function parameter documentation (where the parameter is the `props` object of a particular shape).